src/deferred/exceptionHook.js

```diff
@@ -0,0 +1,21 @@
+define( [
+	"../core",
+	"../deferred"
+], function( jQuery ) {
+
+// These usually indicate a programmer mistake during development,
+// warn about them ASAP rather than swallowing them by default.
+var rerrorNames = /^(Eval|Internal|Range|Reference|Syntax|Type|URI)Error$/;
```

Comment:
The `error.stack` property isn't standardized but it looks like Chrome and Safari do add one for `throw new Error()`. Using that would make this a much noisier hook so I stayed with identifying the ES-defined exception names.